agent/internal/fluent/fluent.go

```diff
@@ -409,38 +409,73 @@ func (f *Fluent) trackLogs(
 
 func removeContainerByName(
 	ctx context.Context,
-	docker *docker.Client,
+	dcl *docker.Client,
 	name string,
 ) error {
-	containers, err := docker.Inner().ContainerList(ctx, types.ContainerListOptions{
+	cont, err := containerByName(ctx, dcl, name)
+	switch {
+	case err != nil:
+		return err
+	case cont == nil:
+		return nil
+	}
+
+	log.WithField("id", cont.ID).Infof("removing Docker container %s", name)
```

Comment:
prob need to backport this removal code to the docker client, but i think that should be a separate pr